matter_server/server/ota/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""OTA implementation for the Matter Server."""
+
+from matter_server.server.ota import dcl
+
+HARDCODED_UPDATES: dict[tuple[int, int], dict] = {
```

Comment:
another approach could be that we create an ABC for an update provider and then have a base implementation for a "hardcoded update" provider. And that when calling the outer 'check_for_update' we check for updates on all providers. This way you also differentiate between main-net and test-net by making test-net a simple subclass (with a few changed params)